### PR TITLE
Plugins Category Results routes now renders the PluginsCategoryResults page.

### LINKED ISSUE #64547: Marketplace: Discovery page route should render the DiscoveryPage component
### Details

Discovery page routes should render the DiscoveryPage component instead of the PluginsBrowser one.

pdh6GB-14m-p2

### Checklist

- [x] `plugins/` and `plugins/:siteId` should render the DiscoveryPage component
- [x] Render the whole page from the DiscoveryPage component.
- [x] Remove unused code from `PluginsBrowser`.

### Related

_No response_

### PR DESCRIPTION
**Pull request by Rolilink**
#### Proposed Changes

* Extracts `PluginsCategoryResults` page from `PluginsBrowser`.
* Categories route now render the `PluginsSearchResults` page.
* Redirects unsupported categories urls to the discovery page.
* Fixes an issue where the search wasn't cleared from the search input when hitting the clear button.

#### Testing Instructions

* Smoke test categories and the results page
* Smoke test the rest of the views (Discovery Page, Search Results Page).
* It should return the correct results
* It should send the correct tracks events
* visiting a url with an unsupported category like `hello` should redirect to the discovery page.

#### Pre-merge Checklist

- [ ] ~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [ ] ~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~

Fixes #64547

### CODE REVIEW

**Comment by Rolilink** [comment]
Requires https://github.com/Automattic/wp-calypso/pull/66668 to be merged

File: client/my-sites/plugins/plugins-browser/clear-search-button.jsx [hunk 1]
```diff
@@ -7,15 +8,20 @@ const ClearSearchButton = () => {
...
+			<span
+				onClick={ () => onClick() }
+				onKeyPress={ () => onClick() }
+				role="link"
+				tabIndex={ 0 }
+				className="plugins-browser__clear-search-button"
 			>
 				{ translate( 'Clear' ) }
-			</a>
+			</span>
```
Comment (lsl) [inline client/my-sites/plugins/plugins-browser/clear-search-button.jsx:24]: Please undo this. Fix in https://github.com/Automattic/wp-calypso/pull/66883

File: client/my-sites/plugins/controller.js [hunk 2]
```diff
@@ -94,6 +101,30 @@ function plugin( context, next ) {
...
+	const category = context.params.category;
+	const site = context.params.site;
+
+	if ( ! ALLOWED_CATEGORIES.includes( category ) ) {
+		const redirectionUrl = `/plugins/${ site || '' }`;
+		page.redirect( redirectionUrl );
+	}
+
+	next();
+}
```
Comment (lsl) [inline client/my-sites/plugins/controller.js:126]: This is meant to be a refactor PR, can we move this behavior change to a new PR?

File: client/my-sites/plugins/plugins-page-view-tracker/index.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+
+const PluginsPageViewTracker = ( { category, selectedSiteId, trackPageViews } ) => {
+	const analyticsPageTitle = 'Plugin Browser' + category ? ` > ${ category }` : '';
+	let analyticsPath = category ? `/plugins/browse/${ category }` : '/plugins';
```
Comment (lsl) [inline client/my-sites/plugins/plugins-page-view-tracker/index.jsx:5]: Can we delete this file and inline the base `<PageViewTracker>` components in each page component with paths specific to each page? Splitting the pages up was supposed to reduce this kind of ternary handling of all the different contexts.

File: client/my-sites/plugins/plugins-search-results-page/index.jsx [hunk 4]
```diff
@@ -93,45 +65,29 @@ const PluginsSearchResultPage = ( {
 				}
 			);
 
-			if ( categoryName ) {
-				title = translate(
-					'Found %(total)s plugin for "%(searchTerm)s" under "%(categoryName)s"',
-					'Found %(total)s plugins for "%(searchTerm)s" under "%(categoryName)s"',
```
Comment (lsl) [inline client/my-sites/plugins/plugins-search-results-page/index.jsx:99]: Why is this being deleted? It was added to get translations in before the functionality was ready. Can we keep this around for the time being?

File: client/my-sites/plugins/search-box-header/index.jsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ const SearchBox = ( { isMobile, searchTerm, doSearch, searchBoxRef, isSearching
 				pinned={ isMobile }
 				fitsContainer={ isMobile }
 				onSearch={ doSearch }
+				onSearchClose={ () => setQueryArgs( {} ) }
```
Comment (lsl) [inline client/my-sites/plugins/search-box-header/index.jsx:23]: Need to rebase against #66883 again I think?

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 6]
```diff
@@ -143,14 +87,22 @@ const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader }
...
 				title={ translate( 'Plugins you need to get your projects done' ) }
 				searchTerms={ [ 'seo', 'pay', 'booking', 'ecommerce', 'newsletter' ] }
 			/>
 
-			{ ! search && <Categories selected={ category } /> }
-			<div className="plugins-browser__main-container">{ renderList() }</div>
-			{ ! category && ! search && <EducationFooter /> }
+			<Categories selected={ category } />
+			<div className="plugins-browser__main-container">
+				<PluginsDiscoveryPage
```
Comment (lsl) [inline client/my-sites/plugins/plugins-browser/index.jsx:96]: Going by `PluginsSearchResultPage` and `PluginsCategoryResultsPage` I expect "Pages" to include a `MainComponent` root. Can we rename `PluginsDiscoveryPage` to something other than `Page`?

**Review by lsl — CHANGES_REQUESTED** [review]
@lupus2k I like where this is going but there is way too much happening in here now. Could we split this up into smaller PR's with clear goals for each? I think we're just trying to do too many refactor steps at once.

File: client/my-sites/plugins/controller.js (hunk 2 above)
Comment (lupus2k) [reply client/my-sites/plugins/controller.js:126]: I'll do it

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts [hunk 7]
```diff
@@ -17,8 +17,7 @@ const selectors = {
 	pluginTitleOnSection: ( section: string, plugin: string ) =>
 		`.plugins-browser-list:has(.plugins-browser-list__title.${ section }) :text-is("${ plugin }")`,
 	sectionTitles: '.plugins-browser-list__title',
-	browseAllFree: 'a[href^="/plugins/browse/popular"]',
-	browseAllPaid: 'a[href^="/plugins/browse/paid"]',
+	browseAllBtn: ( section: string ) => `text=${ section }Browse All >> a`,
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/plugins-page.ts:20]: **suggestion::blocking**: Btn is non-standard terminology in our test suite framework. Please use the term `Button`.

**Review by worldomonation — CHANGES_REQUESTED** [review]
Request change from @Automattic/kitkat end regarding terminology used for the selector.

File: client/my-sites/plugins/controller.js (hunk 2 above)
Comment (lupus2k) [reply client/my-sites/plugins/controller.js:126]: New PR here: #66941

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts (hunk 7 above)
Comment (lupus2k) [reply packages/calypso-e2e/src/lib/pages/plugins-page.ts:20]: Good to know @worldomonation. Thanks!

**Pull request closed without merging**